presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -662,9 +662,21 @@ public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle
 
         Optional<Table> target = metastore.getTable(handle.getSchemaName(), handle.getTableName());
         if (!target.isPresent()) {
```

Comment:
What about skip the `getTable` entirely and remove `if (!target.isPresent())` here? Instead, we could just go issue a drop table regardless, and handle the TableNotFound there.

Reply:
I initially did it this way to keep the old behavior as unchanged as possible, but this makes sense, so updated.